hs_core/models.py

```diff
@@ -4164,6 +4169,8 @@ def update(self, metadata, user):
             for element_name in ('title', 'description', 'language', 'rights'):
                 for dict_item in metadata:
                     if element_name in dict_item:
+                        if element_name in ('title', 'rights') and is_res_published:
```

Comment:
These validations are also being done in the create_element method for the metadata model.  I like the checks being done here better.  Remove the validation checks from the create/update/delete element methods and keep them at this level.

Reply:
Now I realize since I have these validation checks for the published resource already at the CoreMetadata class level (create_element(), update_element(), delete_element()) I don't need these checks here. This update() function is for bulk metadata update and used only by REST call. All metadata editing calls either by REST or front-end passes through the meta editing functions of the CoreMetadata class.

Reply:
Things are so splintered it's hard to keep things straight. This method is used only by the REST API (it is also part of CoreMetadata).  I made an assumption that all the validation rules in this update method were used by the web application but you are right, it goes through a different route and has it's own set of rules.

I don't know the answer here, I'd like the validation to be defined once and used by all routes.  I don't like just the plain is_res_published check and agree with @alvacouch that we could at least group the access control rules into one function instead of implementing pieces of it in different places.